src/unstructured_client/_hooks/custom/split_pdf_hook.py

```diff
@@ -130,43 +136,94 @@ def before_request(
             fallback_value=DEFAULT_CONCURRENCY_LEVEL,
             max_allowed=MAX_CONCURRENCY_LEVEL,
         )
+        limiter = asyncio.Semaphore(concurrency_level)
 
         pdf = PdfReader(io.BytesIO(file.content))
         split_size = get_optimal_split_size(
             num_pages=len(pdf.pages), concurrency_level=concurrency_level
         )
         pages = pdf_utils.get_pdf_pages(pdf, split_size)
 
-        call_api_partial = functools.partial(
-            request_utils.call_api,
-            client=self.client,
-            request=request,
-            form_data=form_data,
-            filename=file.file_name,
-        )
-        self.partition_requests[operation_id] = []
+        async def call_api_partial(page):
+            async with httpx.AsyncClient() as client:
+                status_code, json_response = await request_utils.call_api_async(
+                    client=client,
+                    original_request=request,
+                    form_data=form_data,
+                    filename=file.file_name,
+                    page=page,
+                    limiter=limiter,
+                )
+
+                # convert httpx response to requests.Response to preserve
+                # compatibility with the synchronous SDK generated by speakeasy
+                response = requests.Response()
+                response.status_code = status_code
+                response._content = json.dumps(json_response).encode()  # pylint: disable=W0212
+                response.headers["Content-Type"] = "application/json"
+                return response
+
+        self.coroutines_to_execute[operation_id] = []
         last_page_content = io.BytesIO()
         last_page_number = 0
-        with ThreadPoolExecutor(max_workers=concurrency_level) as executor:
-            for page_content, page_index, all_pages_number in pages:
-                page_number = page_index + starting_page_number
-                # Check if this page is the last one
-                if page_index == all_pages_number - 1:
-                    last_page_content = page_content
-                    last_page_number = page_number
-                    break
-                self.partition_requests[operation_id].append(
-                    executor.submit(call_api_partial, page=(page_content, page_number))
-                )
 
+        for page_content, page_index, all_pages_number in pages:
+            page_number = page_index + starting_page_number
+            # Check if this page is the last one
+            print(f"Page {page_number} of {all_pages_number}")
+            if page_index == all_pages_number - 1:
```

Comment:
@ds-filipknefel you mentioned that there's a bug on the main branch. I'll incorporate your fix on Friday.